linker/shared/src/main/scala/org/scalajs/linker/standard/LinkTimeProperties.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.standard
+
+import org.scalajs.ir.{Types => jstpe, Trees => js}
+import org.scalajs.ir.ScalaJSVersions
+import org.scalajs.ir.Position.NoPosition
+import org.scalajs.linker.interface.{Semantics, ESFeatures}
+
+final class LinkTimeProperties private[standard] (
+  semantics: Semantics,
+  esFeatures: ESFeatures,
+  targetIsWebAssembly: Boolean
+) {
+  import LinkTimeProperties._
+  import LinkTimeValue._
+
+  private val linkTimeProperties: Map[String, LinkTimeValue] = Map(
+    "core/esVersion" ->
+      LinkTimeValue.LinkTimeInt(esFeatures.esVersion.edition),
+    "core/assumingES6" ->
+      LinkTimeValue.LinkTimeBoolean(esFeatures.useECMAScript2015Semantics),
+    "core/isWebAssembly" ->
+      LinkTimeValue.LinkTimeBoolean(targetIsWebAssembly),
+    "core/productionMode" ->
+      LinkTimeValue.LinkTimeBoolean(semantics.productionMode),
+    "core/linkerVersion" ->
+      LinkTimeValue.LinkTimeString(ScalaJSVersions.current),
+  )
+
+  def transformLinkTimeProperty(prop: js.LinkTimeProperty): js.Literal = {
+    val value = linkTimeProperties.getOrElse(prop.name,
+        throw new IllegalArgumentException(s"link time property not found: '$prop'"))
+    // TODO: typecheck
+    // println(s"transofrmLinkTimeProperty: $prop of type ${prop.tpe}")
```

Comment:
Since we cannot carry the type information from `linkTimeProperty[T](...)` because of type erasure (as mentioned in https://github.com/scala-js/scala-js/pull/5000#issuecomment-2308905997), unable to check the transformed `Literal` type is `T` :(